### PR TITLE
store: reject stale RequestVote.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay

### CODE REVIEW

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -246,6 +246,19 @@ impl<T: Transport, C: PdClient> Store<T, C> {
...
         }
 
+        // Should we need to check other types for stale message?
+        if msg.get_message().get_msg_type() == MessageType::MsgRequestVote {
+            if let Some(peer) = self.region_peers.get(&region_id) {
+                let from_epoch = msg.get_region_epoch();
+                // Is it necessary to check version too?
+                if from_epoch.get_conf_ver() <
+                   peer.storage.rl().region.get_region_epoch().get_conf_ver() {
+                    warn!("RequestVote with a stale epoch {:?}, reject", from_epoch);
```
Comment (ngaut) [inline src/raftstore/store/store.rs]: s/reject/ignore it/

**Comment by ngaut** [comment]
Rest LGTM.

**Comment by ngaut** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Comment by BusyJay** [comment]
Maybe we should change `reject` to `drop`.

**Comment by siddontang** [comment]
I think reject and drop both are ok.

**Pull request merged**